podcasts/Whats New/WhatsNew.swift

```diff
@@ -38,6 +36,26 @@ class WhatsNew {
 
         return whatsNewViewController
     }
+
+    /// Returns the announcement to be displayed if one is available
+    var visibleAnnouncement: Announcement? {
+        // Don't show any announcements if this is the first run of the app,
+        // or if we've already checked the what's new for this version
+        guard let previousOpenedVersion, previousOpenedVersion != currentVersion else {
+            return nil
+        }
+
+        // Find the last announcement that:
+        // - is enabled
+        // - has not been shown already
+        // - the target version is not before the last opened version, and not for a future version
+        return announcements
+            .last(where: {
+                $0.isEnabled &&
```

Comment:
If we change the `isEnabled` to be an `@autoclosure` we can move this to be the last check. This would mean that the closure would only run if the announcement could be displayed, so we wouldn't need to check it for all announcements.